addons/mail/models/mail_thread.py

```diff
@@ -1735,13 +1737,15 @@ def _message_post_process_attachments(self, attachments, attachment_ids, message
                 cid = False
                 if len(attachment) == 2:
                     name, content = attachment
+                    info = {}
                 elif len(attachment) == 3:
                     name, content, info = attachment
                     cid = info and info.get('cid')
                 else:
                     continue
                 if isinstance(content, str):
-                    content = content.encode('utf-8')
+                    encoding = info and info.get('encoding', False)
+                    content = content.encode(encoding or 'utf-8')
```

Comment:
100% nitpicking, feel free to ignore. `False` is an artifact of the Odoo ORM, it exists only because `None` cannot be serialized in XMLRPC (without enabling some extensions) thus when something comes/goes from/to XMLRPC, all `None` should be converted to `False` (which is bevahior-wise, very close to `None`). It is not the case here, we can safely use `None` which is `get`'s default default.
```suggestion
                    encoding = info and info.get('encoding')
                    content = content.encode(encoding or 'utf-8')
```

Reply:
@Julien00859 Indeed, thanks! Updated :+1: